### PR TITLE
8340271: Open source several AWT Robot tests

### PR DESCRIPTION
**Pull request by prsadhuk**
Opensource few Robot tests

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8340271](https://bugs.openjdk.org/browse/JDK-8340271): Open source several AWT Robot tests (**Bug** - P4)

### Reviewers
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - **Reviewer**)
 * [Harshitha Onkar](https://openjdk.org/census#honkar) (@honkar-jdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21062/head:pull/21062` \
`$ git checkout pull/21062`

Update a local copy of the PR: \
`$ git checkout pull/21062` \
`$ git pull https://git.openjdk.org/jdk.git pull/21062/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21062`

View PR using the GUI difftool: \
`$ git pr show -t 21062`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21062.diff">https://git.openjdk.org/jdk/pull/21062.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21062#issuecomment-2358485634)

### CODE REVIEW

File: test/jdk/java/awt/Robot/RobotScrollTest.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import java.awt.TextArea;
+
+public class RobotScrollTest {
+
+    static TextArea ta;
+    static Robot robot;
+
+    private static final String INSTRUCTIONS = """
+         0. DON'T TOUCH ANYTHING!
+         1. This test is for Win32 and Linux only.
```
Comment (honkar-jdk) [inline test/jdk/java/awt/Robot/RobotScrollTest.java:47]:
```suggestion
         1. This test is for Win and Linux only.
```

File: test/jdk/java/awt/Robot/RobotScrollTest.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        doTest();
+    }
+
+    private static void doTest() {
+        robot.waitForIdle();
+        robot.delay(1000);
+        // get loc of TextArea
+        Point taAt = ta.getLocationOnScreen();
+        // get bounds of button
+        Rectangle bounds = ta.getBounds();
```
Comment (honkar-jdk) [inline test/jdk/java/awt/Robot/RobotScrollTest.java:82]: Should they be in EDT?

File: test/jdk/java/awt/Robot/RobotScrollTest.java (hunk 1 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Robot/RobotScrollTest.java]: done

File: test/jdk/java/awt/Robot/RobotScrollTest.java (hunk 2 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Robot/RobotScrollTest.java:82]: manual test, not needed

File: test/jdk/java/awt/Robot/CreateScreenCapture.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import java.awt.Frame;
+import java.awt.Image;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.TextArea;
+
+public class CreateScreenCapture {
+
+    static Robot robot;
+    static Dialog dialog;
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Robot/CreateScreenCapture.java:44]: may be declared as local variable.

File: test/jdk/java/awt/Robot/CreateScreenCapture.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+         After 5-7 steps the size taken by the process should become stable.
+         If this happens, then test passed otherwise test failed.
+
+         Small oscillations of the memory size are, however, acceptable.""";
+
+    public static void main(String[] args) throws Exception {
+        robot = new Robot();
+        PassFailJFrame passFail = new PassFailJFrame(INSTRUCTIONS);
+        dialog = new Dialog(new Frame(), "Instructions");
+        messageText = new TextArea( "", 5, 80, TextArea.SCROLLBARS_BOTH);
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Robot/CreateScreenCapture.java:62]:
```suggestion
        messageText = new TextArea("", 5, 80, TextArea.SCROLLBARS_BOTH);
```

File: test/jdk/java/awt/Robot/CreateScreenCapture.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        PassFailJFrame.positionTestWindow(dialog, PassFailJFrame.Position.HORIZONTAL);
+        dialog.setSize(200, 300);
+        dialog.setVisible(true);
+        Rectangle rect = new Rectangle(0, 0, 1000, 1000);
+        for (int i = 0; i < 100; i++) {
+            Image image = robot.createScreenCapture(rect);
+            image.flush();
+            image = null;
+            robot.delay(200);
+            log("step #"+i);
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Robot/CreateScreenCapture.java:74]:
```suggestion
            log("step #" + i);
```

File: test/jdk/java/awt/Robot/CreateScreenCapture.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        }
+        passFail.awaitAndCheck();
+    }
+
+    private static void log(String messageIn) {
+        messageText.append( messageIn + "\n" );
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Robot/CreateScreenCapture.java:80]:
```suggestion
        messageText.append(messageIn + "\n");
```

File: test/jdk/java/awt/Robot/RobotScrollTest.java [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    static TextArea ta;
+    static Robot robot;
+
+    private static final String INSTRUCTIONS = """
+         0. DON'T TOUCH ANYTHING!
+         1. This test is for Windown and Linux only.
+         2. Just sit back, and watch the Robot move the mouse to the TextArea.
+         3. Once the pointer is on the text area, the Robot will use the mouse wheel
+            to scroll the text.
+            If the text scrolled, hit PASS, else, hit fail.""";
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Robot/RobotScrollTest.java:51]:
```suggestion
            If the text scrolled, hit PASS, else, hit FAIL.""";
```

File: test/jdk/java/awt/Robot/RobotScrollTest.java [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+                        taAt.y + bounds.height / 2);
+
+        // rotate wheel a few times
+        for (int j = 1; j < 8; j++) {
+            for (int k = 0; k < 5; k++) {
+                robot.mouseWheel(j);
+            }
+
+            for (int k = 0; k < 5; k++) {
+                robot.mouseWheel(-1*j);
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Robot/RobotScrollTest.java:95]:
```suggestion
                robot.mouseWheel(-1 * j);
```

File: test/jdk/java/awt/Robot/CreateScreenCapture.java [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+import java.awt.Dialog;
+import java.awt.Frame;
+import java.awt.Image;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.TextArea;
+
+public class CreateScreenCapture {
+
+    static Robot robot;
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Robot/CreateScreenCapture.java:43]: `robot` var also can be moved inside main.

File: test/jdk/java/awt/Robot/CreateScreenCapture.java [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+            image.flush();
+            image = null;
+            robot.delay(200);
+            log("step #" + i);
+        }
+        passFail.awaitAndCheck();
+    }
+
+    private static void log(String messageIn) {
+        messageText.append( messageIn + "\n");
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Robot/CreateScreenCapture.java:79]:
```suggestion
        messageText.append(messageIn + "\n");
```

File: test/jdk/java/awt/Robot/RobotScrollTest.java [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+import java.awt.TextArea;
+
+public class RobotScrollTest {
+
+    static TextArea ta;
+    static Robot robot;
+
+    private static final String INSTRUCTIONS = """
+         0. DON'T TOUCH ANYTHING!
+         1. This test is for Windown and Linux only.
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Robot/RobotScrollTest.java:47]:
```suggestion
         1. This test is for Windows and Linux only.
```

**Review by kumarabhi006 — APPROVED** [review]

**Review by honkar-jdk — APPROVED** [review]

**Pull request closed without merging**